src/grammar/grammar.ohm

```diff
@@ -263,12 +263,22 @@ Tact {
     id = ~reservedWord #idStart #(idPart*)
 
     // FunC identifiers, where `funcId` stands for FunC function identifier
-    // The plain identifier cannot be a single underscore nor just a number
-    funcPlainId = "_"? "-"? "0x"? hexDigit* ~hexDigit (~(whiteSpace | "(" | ")" | "," | "." | ";" | "~") any)+
+    // A plain identifier cannot be a number or a single underscore
+    funcPlainId = ("-"? "0x" hexDigit+ ~hexDigit)? ("-"? digit+ ~digit)? "_"? (~(whiteSpace | "(" | ")" | "," | "." | ";" | "~") any)+ --generic
+                | "-"? "0x" hexDigit* "_" --justHexUnderscore
+                | "-"? digit+ "_" --justDecUnderscore
 
     funcQuotedId = "`" (~("`" | "\n") any)+ "`"
 
-    funcId = ~("\"" | "{-") ("." | "~")? (funcQuotedId | funcPlainId)
+    funcId = ~("\"" | "{-" | "#") ("." | "~")? (funcQuotedId | funcPlainId)
+
+    /*
+      NOTE(novusnota):
+
```

Comment:
```suggestion
      NOTE(novusnota):

```

Reply:
Let's remove this part, there is commit history to query authorship and additional context